### PR TITLE
Options window cleanup

### LINKED ISSUE #4733: Thoughts on options menu
> The Interface and Window tabs of the options menu were merged recently because the tab row got too wide, so it > seems rather cluttered now. Suggestions how to unclutter it without adding more tabs welcome…

_Originally posted by @Noordfrees in https://github.com/widelands/wl_addons_server/issues/23#issuecomment-786811833_

Edit: Will ad some of my thoughts in the next days

### PR DESCRIPTION
**Pull request by tothxa**
**Type of change**
Cleanup

**Issue(s) closed**
Fixes #4733

Also fixed a bug in `Statebox::layout()` that prevented setting the width of checkboxes to exactly fit the text.

**Possible regressions**
 * Setting options, especially writing syncstreams and mouse grabbing issues.
 * Checkbox layouting.

**Additional context**
The translation stats space fix has some ugly workarounds for issues like #6036

### CODE REVIEW

**Comment by Noordfrees** [comment]
I don't like the changed sound options. With the new design it is no longer intuitive which slider belongs to which checkbox because the distance is so huge. The sliders just seem to be hanging in the middle of nothing with no visual connection to the labels or checkboxes.

grafik

Other than that, tested and working, mouse grab and syncstreams behaviour as expected. I like the cleaner General options tab :)

**Comment by tothxa** [comment]
> I don't like the changed sound options. With the new design it is no longer intuitive which slider belongs to which checkbox because the distance is so huge. The sliders just seem to be hanging in the middle of nothing with no visual connection to the labels or checkboxes.

I made the label checkboxes adapt to the biggest actual text width. Turns out checkboxes had a bug in the `layout()` function that prevented fitting them closely, so I fixed that too. Please check if it's better for you this way.

It still doesn't work right for RTL languages, but I think that's a bug in the `text_width()` function (in `graphic/text_layout`), probably due to using LTR fonts for the missing translations. So out of scope here.

> Other than that, tested and working, mouse grab and syncstreams behaviour as expected. I like the cleaner General options tab :)

Well, not for long if I finish #5944 ;)

**Comment by Noordfrees** [comment]
That's better but still looks somewhat untidy. Since the tab has a lot of unused space, perhaps you could increase the spacing between the rows to set them more apart, like this:
grafik

Also, another idea: Instead of disabling the sliders for deactivated sound how about hiding them completely and showing sliders only for enabled sound types?

**Comment by tothxa** [comment]
> That's better but still looks somewhat untidy. Since the tab has a lot of unused space, perhaps you could increase the spacing between the rows to set them more apart
> 
> Also, another idea: Instead of disabling the sliders for deactivated sound how about hiding them completely and showing sliders only for enabled sound types?

Both done. Though hiding sliders is even worse for RTL, because the checkbox jumps to the left when a type is disabled. Again, I tried to solve it, but gave up. (I guess I could work it around again with extra padding panels, but I didn't want to.)

File: src/wlapplication.cc [hunk 1]
```diff
@@ -465,7 +465,16 @@ WLApplication::WLApplication(int const argc, char const* const* const argv)
 	// register it once.
 	UI::Panel::register_click();
 
-	set_input_grab(get_config_bool("inputgrab", false));
+	// Disable mouse grabbing
+	// TODO(tothxa): keep around until tested without it
```
Comment (Noordfrees) [inline src/wlapplication.cc]: tested, this can go

**Review by Noordfrees — COMMENTED** [review]
Sound options look good to me now :)

Code LGTM

**Review by Noordfrees — APPROVED** [review]

**Comment by tothxa** [comment]
Thank you. :)

**Pull request merged**